src/transformation/visitors/switch.ts

```diff
@@ -55,15 +55,25 @@ export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (st
 
     // If the switch only has a default clause, wrap it in a single do.
     // Otherwise, we need to generate a set of if statements to emulate the switch.
-    let statements: lua.Statement[] = [];
+    const statements: lua.Statement[] = [];
+    const prefixStatements: lua.Statement[] = [];
```

Comment:
Think it might be clearer if you just named this `hoistedStatements`? I think to avoid naming conflicts with your destructuring assignments is to just not destructure, I don't think that is really making things clearer when it's split over multiple lines like this anyway.